File: PHP/Day 1 - Sonar Sweep Part Two.php
<?php
/**
 * Created by PhpStorm.
 * User: angelgrigorov
 * Date: 01.12.21
 * Time: 12:12
 */

$depthMeasurement = intval(readline());
$allDepthMeasurements = [];
while(true){
    $allDepthMeasurements[] = $depthMeasurement;

    if($depthMeasurement == "6624"){
        break;
    }
    $depthMeasurement = intval(readline());
}
$summedDepthMeasurements = [];
for($i = 0; $i < count($allDepthMeasurements) - 2; $i++){
    $summedDepthMeasurements[] = $allDepthMeasurements[$i] + $allDepthMeasurements[$i + 1] + $allDepthMeasurements[$i + 2];
}

$lastDepthMeasurement = 0;
$count = 0;

for($j = 0; $j < count($summedDepthMeasurements); $j++){
    if($lastDepthMeasurement != 0 && $summedDepthMeasurements[$j] > $lastDepthMeasurement){
        $count++;
    }
    $lastDepthMeasurement = $summedDepthMeasurements[$j];
}

echo $count;

File: GoLang/Day 1 - Sonar Sweep Part Two.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	file, _ := os.Open("dayOneInput.txt")
	defer file.Close()
	scanner := bufio.NewScanner(file)
	count := 0
	lastDepthMeasurement := 0
	var allDepthMeasurements []int
	var summedDepthMeasurements []int
	for scanner.Scan() {
		depthMeasurement, _ := strconv.Atoi(scanner.Text())
		allDepthMeasurements = append(allDepthMeasurements, depthMeasurement)
	}
	for i := 0; i < len(allDepthMeasurements)-2; i++ {
		summedDepthMeasurements = append(summedDepthMeasurements, allDepthMeasurements[i]+allDepthMeasurements[i+1]+allDepthMeasurements[i+2])
	}
	for j := 0; j < len(summedDepthMeasurements); j++ {
		if lastDepthMeasurement != 0 && summedDepthMeasurements[j] > lastDepthMeasurement {
			count++
		}
		lastDepthMeasurement = summedDepthMeasurements[j]
	}
	fmt.Println(count)
}
